### ISSUE TITLE
Switch to serde when possible

### ISSUE THREAD

**Issue by davidpodhola** [hsharpsoftware/rust-realworld-example-app#27]
Blocked by [nickel.rs-Support Serde](https://github.com/nickel-org/nickel.rs/issues/345)

**Comment by dtolnay** [comment]
Looks like Nickel has been abandoned. Maybe there is an alternative web framework that is better maintained?

**Comment by davidpodhola** [comment]
Hm, that is not nice to hear. I chose Nickel mainly because I found [Build an API in Rust with JWT Authentication using Nickel.rs](https://auth0.com/blog/build-an-api-in-rust-with-jwt-authentication-using-nickelrs/) and then  [nickel-jwt-session](https://github.com/kaj/nickel-jwt-session) very soon. 

Anyway I assume the best would be [iron](https://github.com/iron/iron). Or do you have any other recommendation? Thanks!

**Comment by dtolnay** [comment]
Haha that is an extremely old blog post in Rust terms.

I haven't used Nickel or Iron but Iron's primary maintainer has been very unresponsive in my experience. I would encourage you to consult [this graph](https://github.com/iron/iron/graphs/contributors) before choosing a project.

I think [Hyper](https://hyper.rs/) is the current preference? It is definitely well maintained, has 5x as many downloads as Iron, and has 7x as many crates depending on it.

**Comment by davidpodhola** [comment]
Yes, you are right, it is clear things are moving fast in Rust world :-)

Thanks for your insights. As this project should serve somehow also for comparisons between different languages, platforms etc. I think it is important to be on the latest technologies.

I also do not have problems with these evolving over time, in fact I find this helpful.

Thanks again!

**Comment by davidpodhola** [comment]
Tracked in #29, already some of the endpoints implemented and is very nice!

**Comment by davidpodhola** [comment]
Merged into master, from now on use [Hyper](https://hyper.rs/).
